### PR TITLE
Use SortOrder from shims.

### PR DESCRIPTION
**Pull request by firestarman**
Use SorOrder constructor from shims to fix the test failures about udf for spark 3.1.0.

The error log:
```
>   format(target_id, ".", name), value)
 E  py4j.protocol.Py4JJavaError: An error occurred while calling o2148.collectToPython.
 E   : java.lang.NoSuchMethodError: org.apache.spark.sql.catalyst.expressions.SortOrder$.apply$default$3()Lscala/collection/immutable/Set;
 E   at org.apache.spark.sql.rapids.execution.python.GpuFlatMapCoGroupsInPandasExec.$anonfun$requiredChildOrdering$1(GpuFlatMapCoGroupsInPandasExec.scala:101)
 E   at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:238)
```

More details pls refer to the PR https://github.com/NVIDIA/spark-rapids/pull/1232  

Signed-off-by: Firestarman <firestarmanllc@gmail.com>

### CODE REVIEW

**Comment by firestarman** [comment]
build

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/python/GpuFlatMapCoGroupsInPandasExec.scala [hunk 1]
```diff
@@ -98,7 +98,9 @@ case class GpuFlatMapCoGroupsInPandasExec(
 
   override def requiredChildOrdering: Seq[Seq[SortOrder]] = {
     leftGroup
-      .map(SortOrder(_, Ascending)) :: rightGroup.map(SortOrder(_, Ascending)) :: Nil
+      .map(ShimLoader.getSparkShims.sortOrder(_, Ascending)) ::
+    rightGroup
+      .map(ShimLoader.getSparkShims.sortOrder(_, Ascending)) :: Nil
```
Comment (pxLi) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/python/GpuFlatMapCoGroupsInPandasExec.scala:103]: nit: 2 space indent for line wrapping https://docs.scala-lang.org/style/indentation.html#line-wrapping

Comment (firestarman) [reply sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/python/GpuFlatMapCoGroupsInPandasExec.scala:103]: Which one should it be ?
```
    leftGroup
      .map(SortOrder(_, Ascending)) :: rightGroup.map(SortOrder(_, Ascending)) :: Nil
      rightGroup
        .map(ShimLoader.getSparkShims.sortOrder(_, Ascending)) :: Nil
```
or 
```
    leftGroup
      .map(SortOrder(_, Ascending)) :: rightGroup.map(SortOrder(_, Ascending)) :: Nil
        rightGroup
          .map(ShimLoader.getSparkShims.sortOrder(_, Ascending)) :: Nil
```

Comment (pxLi) [reply sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/python/GpuFlatMapCoGroupsInPandasExec.scala:103]: should be fine w/ just 
```
    leftGroup.map(ShimLoader.getSparkShims.sortOrder(_, Ascending)) ::
      rightGroup.map(ShimLoader.getSparkShims.sortOrder(_, Ascending)) :: Nil
```

**Comment by firestarman** [comment]
build

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/python/GpuFlatMapCoGroupsInPandasExec.scala (hunk 1 above)
Comment (firestarman) [reply sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/python/GpuFlatMapCoGroupsInPandasExec.scala]: done

**Review by pxLi — APPROVED** [review]
LGTM

**Comment by firestarman** [comment]
> LGTM

Then you can resolve that conversation.

**Pull request merged**